packages/devtools_app/lib/src/debugger/hover.dart

```diff
@@ -8,52 +8,53 @@ import 'package:flutter/material.dart';
 
 import '../theme.dart';
 
+/// Regex for valid Dart identifiers.
+final _identifier = RegExp(r'^[a-zA-Z0-9]|_|\$');
+
 /// Returns the word in the [line] for the provided hover [dx] offset given
 /// the [line]'s [textStyle].
 String wordForHover(double dx, TextSpan line, TextStyle textStyle) {
-  final index = _hoverIndex(dx, line, textStyle);
-  var word = '';
-  if (index == -1) return word;
-  word = line.children[index].toPlainText();
-  word = _mergedLinkedWords(word, index, line);
-  return word;
-}
+  final painter = TextPainter(
+    text: TextSpan(children: [line], style: textStyle),
+    textDirection: TextDirection.ltr,
+  )..layout();
+  final lineText = line.toPlainText();
+  final hoverIndex = ((dx / painter.width) * (lineText.length - 1)).round();
```

Comment:
try to avoid assuming letters have fixed width. We already know that isn't accurate for our currently styled text (unclear exactly why). Better to instead measure the length up until each point so we are exactly right at the hover location.

Reply:
Discussed offline. Using a slightly different approach.